src/chainstate/burn/operations/delegate_stx.rs

```diff
@@ -32,13 +33,13 @@ impl DelegateStxOp {
         )
     }
 
-    fn parse_data(data: &Vec<u8>) -> Option<ParsedData> {
+    fn parse_data(data: &mut Vec<u8>) -> Option<ParsedData> {
```

Comment:
This is because I call the function `truncate` on it. 

I'll decide how to proceed with this based on the other discussion in this review on truncation.

Reply:
Ah.  I don't think the burnchain parser allows the payload to be longer than 77 bytes as it is.  But even if it does (or will do so in a subsequent release), it's probably better to just do this instead of mutating the argument:

```
let relevant_data = &data[0, cmp::max(77, data.len())]
// do stuff with relevant_data instead of data.
```

Reply:
Yeah, I agree with the above -- I'd prefer either adding a new slice variable like @jcnelson recommends or shadowing the existing one:

```rust
let data = &data[..cmp::min(77, data.len()];
```

(Note: I think you want cmp::min here, not max!)